docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+
+```{code-cell} ipython3
+%pip install jupysql --upgrade duckdb-engine matplotlib ipywidgets --quiet
+```
+
+## Import Libraries
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+from sql.ggplot import ggplot, aes, geom_boxplot, geom_histogram, facet_wrap
+import ipywidgets as widgets
+from ipywidgets import interact
+```
+
+## Load the data
+
+```{important}
+<b>Note:</b> If you are following these lessons locally and <b>not</b> on Google Colab, then there is no need to load the data again.
+```
+
+This section was covered in detail in the previous tutorial: [Joining Data in SQL](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#load-the-data). We will be using the same data in this tutorial as well.
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+import sys
+
+sys.path.insert(0, "../../")
+
+import banking  # noqa: E402
+
+
+_ = banking.MarketData("https://tinyurl.com/jb-bank-m", "expanded_data")
+_.extract_asc_to_csv()
+```
+
+If you ran the above cell, you should have a folder `expanded_data` in your current directory that contains the `.csv` files we will be using. In this tutorial, we will be focusing on three of these files: `loan.csv`, `account.csv`, `district.csv`.
+
+## Load Engine
+
+We now load in our SQL extension that allows us to execute SQL queries in Jupyter Notebooks.
+
+```{important}
+<b>Note</b> Ensure you restart any previous notebook that has the same database name as the one initialized below.
+```
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%load_ext sql
+# Initiating a DuckDB database named 'bank_data.duck.db' to run SQL queries
+%sql duckdb:///bank_data.duck.db
+```
+
+<!-- endregion -->
+
+## Creating Tables
+
+Let's start off with loading three of the eight `.csv` files from the `expanded_data` folder in the current directory to our newly created DuckDB database. Like in the previous tutorial, we will [create a schema](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#creating-a-schema) `s1` in which we will store the tables. Here we use the `CREATE TABLE` syntax in DuckDB to ingest four of the eight `.csv` files. The `read_csv_auto` is a function that helps SQL understand our local `.csv` file for creation into our database.
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+The code above will create three tables in the database schema: `s1.account`, `s1.district`, `s1.loan`.
+
+## Exploring the data
+
+Let's take a look at its entries.
+
+```{code-cell} ipython3
+%sqlcmd explore --table s1.account
+```
+
+```{code-cell} ipython3
```

Comment:
For some reason exploring s1.district and s1.loan do not show on the build doc. Does this occur on your end?

Reply:
This is quite buggy. For this PR, in my build the docs, I don't get an output for `s1.account` in the ggplot doc and in the plotly doc. In the seaborn PR (seaborn doc), I don't get an output for `s1.district`. In the same PR but in the plotly doc, I get an output for all except `s1.account`.

Moreover, I have noticed that when I am running all cells in Jupyter Lab, most often than not one of the three `explore` cells don't output anything. So I have to manually run them later. @lfunderburk any idea about these issues?

Reply:
Can you submit an issue on the JupySQL repo on this behaviour.